addons/towing/functions/fnc_detachRope.sqf

```diff
@@ -0,0 +1,63 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Dystopian
+ * Detaches rope of given hook and gives rope item back.
+ *
+ * Arguments:
+ * 0: Player <OBJECT>
+ * 1: Rope Hook <OBJECT>
+ * 2: Deleted object <OBJECT> (default: objNull)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, cursorObject] call ace_towing_fnc_detachRope
+ *
+ * Public: No
+ */
+params ["_unit", "_hook", ["_deletedObject", objNull]];
+
+private _hookVars = _hook getVariable QGVAR(vars);
+if (isNil "_hookVars") then { // this is hookParent
+    _hook = _hook getVariable QGVAR(hook);
+    _hookVars = _hook getVariable QGVAR(vars);
+};
+
+_hookVars params ["_parent", "_child", "_rope", "_ropeClass", "_hookParent"];
+
+TRACE_7("detachRope",_unit,_parent,_child,_hook,_hookParent,_rope,_deletedObject);
+
+ropeDestroy _rope; // can run on client
+
+if (!isNull _unit) then {
+    TRACE_1("rope",_ropeClass);
+    if (!isNull _unit && {_ropeClass isNotEqualTo ""}) then {
+        [_unit, _ropeClass, true] call CBA_fnc_addItem;
+    };
+};
+
+{
+    detach _x;
+    deleteVehicle _x;
+} forEach [_hook, _hookParent];
+
+// cleanup object variables and EHs only if function isn't called from Deleted EH
+if (isNull _deletedObject || {_parent isNotEqualTo _deletedObject}) then {
```

Comment:
Just to make it consistent with the other instances:
```suggestion
if (isNull _deletedObject || {_parent != _deletedObject}) then {
```

Reply:
We deal here with object which has just been deleted. I can't test carefully your code next 2 weeks while I was testing my code since I created the PR. So sorry I can't accept this change until testing.